layer.md

```diff
@@ -1,63 +1,129 @@
-# Creating an Image Filesystem Changeset
+# Filesystem Layer Changeset
 
-An example of creating an Image Filesystem Changeset follows.
+A layer changeset is the distributable difference between filesystems.
+A layer changeset is not directly extracted, but applied to a filesystem.
 
-An image root filesystem is first created as an empty directory.
-Here is the initial empty directory structure for a changeset using the randomly-generated directory name `c3167915dc9d` ([actual layer DiffIDs are generated based on the content](#id_desc)).
+## Distributable Format
+
+Layer Changesets for the [mediatype](./media-types.md) `application/vnd.oci.image.layer.tar+gzip` MUST be packaged in [tar archive][tar-archive].
+Layer Changesets for the [mediatype](./media-types.md) `application/vnd.oci.image.layer.tar+gzip` MUST NOT include duplicate entries for file paths in the resulting [tar archive][tar-archive].
+
+## Change Types
+
+Types of changes that can occur in a changeset are:
+
+* Additions
+* Modifications
+* Removals
+
+Additions and Modifications are reflected the same in the changeset tar archive.
+
+Removals are reflected using "[whiteout](#whiteouts)" file entries (See [Representing Changes](#representing-changes)).
+
+### File Attributes
+
+File attributes that are provided for Additions and Modifications include:
+
+* Modification Time (`mtime`)
+* User ID (`uid`)
+ * User Name (`uname`) is secondary to `uid`
+* Group ID (`gid `)
+ * Group Name (`gname`) is secondary to `gid`
+* Mode (`mode`)
+* Extended Attributes (`xattrs`)
```

Comment:
On Wed, Sep 07, 2016 at 06:58:28AM -0700, Vincent Batts wrote:
> There is no particular tar spec.

Hmm, it looks like there was a run at trying to get a formal tar media
type in 2014 [1], but I don't see it listed by the IANA [2].  Without
such a spec, validating layer blobs is going to be tricky (Alice: “My
tar library cannot parse your layer!”  Bob: “So what?”).  In the
absence of a tar spec, I see two choices:

a. Punt the specifics to users [3].  Lots of freedom, but
   application/vnd.oci.image.layer.tar+gzip becomes unenforceble and
   there are likely portablility edge cases (e.g. sparse files [4]).

b. Require a specified format like pax [5].  Implementers have to
   figure out how to get their “tar” library to write pax, or have to
   use a pax-specific library.  But
   application/vnd.oci.image.layer.pax+gzip is enforcable and portable
   between all image-spec implementations.

[1]: https://www.ietf.org/mail-archive/web/media-types/current/msg00592.html
[2]: http://www.iana.org/assignments/media-types/media-types.xhtml#application
[3]: https://github.com/opencontainers/image-spec/pull/255#discussion_r77852725
[4]: https://github.com/opencontainers/image-spec/pull/255#discussion_r77822800
[5]: http://pubs.opengroup.org/onlinepubs/9699919799/utilities/pax.html#tag_20_92_13